internal/backends/java/msch.go

```diff
@@ -46,18 +50,45 @@ type SearchResult struct {
 	} `json:"response"`
 }
 
-func Search(keyword string) ([]SearchDoc, error) {
-	searchUrl := mavenURL + url.QueryEscape(keyword)
-	res, err := http.Get(searchUrl)
+func mavenSearch(searchURL string) ([]SearchDoc, error) {
+	res, err := http.Get(searchURL)
 	if err != nil {
 		return []SearchDoc{}, err
 	}
 	defer res.Body.Close()
-	decoder := json.NewDecoder(res.Body)
-	var searchResult SearchResult
-	err = decoder.Decode(&searchResult)
+
+	body, err := ioutil.ReadAll(res.Body)
 	if err != nil {
+		fmt.Printf("Could not read response\n")
+		return []SearchDoc{}, err
+	}
+
+	var searchResult SearchResult
+	if err := json.Unmarshal(body, &searchResult); err != nil {
+		fmt.Printf("Failed to decode response %q\n", body)
 		return []SearchDoc{}, err
 	}
+
 	return searchResult.Response.Docs, nil
 }
+
+func Search(keyword string) ([]SearchDoc, error) {
+	searchURL := mavenURL + url.QueryEscape(keyword)
+
+	return mavenSearch(searchURL)
+}
+
+func Info(name string) (SearchDoc, error) {
+	parts := strings.Split(string(name), ":")
```

Comment:
Can you make this resilient to badly-formatted names? For example, running `upm -l java info blade-mvc` produces a stack trace because I've left out a group name.

Reply:
Yeah, let me see what I can do.

Reply:
Changed the implementation to treat input without a group as artifact only search.